### PR TITLE
Adding function to generate Observable charts

### PR DESCRIPTION
**Pull request by nshiab**
We can now call saveChart() on the SimpleData class to save a chart (dot, line, bar, box) into an HTML file.

When we'll have two classes (one for browser and one for one), we will need to make sure this method is only available in node. It will be a different one for the browser.

### CODE REVIEW

File: types/jsdom.d.ts [hunk 1]
```diff
@@ -0,0 +1 @@
+declare module 'jsdom';
```
Comment (isabellebouchard) [inline types/jsdom.d.ts:1]: Have you tried installing the package types definition?
`npm -i --save-dev @types/jsdom`

If they are available, then we do not need this file. If they are not available, then we should declare the package here.

File: types/observablehq-plot.d.ts [hunk 2]
```diff
@@ -0,0 +1 @@
+declare module '@observablehq/plot';
```
Comment (isabellebouchard) [inline types/observablehq-plot.d.ts:1]: same as above :)

**Review by isabellebouchard — APPROVED** [review]
I left two small comments about librairies types definition. 

Good to go once this is fixed! 🚀  Very nice feature!

**Review by isabellebouchard — APPROVED** [review]

File: src/methods/saveCustomChart.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+    const plot = observablePlot.outerHTML
+
+    fs.writeFileSync(
+        path,
+        plot.replace("<svg", '<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink"')
+    )
+
+    return data
+}
```
Comment (isabellebouchard) [inline src/methods/saveCustomChart.ts:28]: There is a lot of code duplicate between this function and `saveChart`. 
`saveChart` should probably build its plot and pass it to this function.

**Review by isabellebouchard — COMMENTED** [review]
I would put some more love into `saveCustomChart` and `saveChart` to avoid code duplicates. What do you think?

File: test/integration/test.js [hunk 4]
```diff
@@ -144,4 +144,22 @@ simpleData
...
+            grid: true,
+            facet: {
+                data: simpleData.data,
+                y: "unit"
+            },
+            marks: [
+                Plot.dotX(simpleData.data, { x: "salary", fill: "unit" })
+            ]
+        })
+    )
```
[the remark this answers is not in the document]
Comment (nshiab) [reply test/integration/test.js:165]: Hum. I think I understood what you said. I'll commit a modification.

**Pull request merged**